elyra/pipeline/component.py

```diff
@@ -277,12 +277,13 @@ def _log_warning(msg: str, logger: Optional[Logger] = None):
 
 
 class ComponentParser(LoggingConfigurable):  # ABC
-    _component_platform: RuntimeProcessorType = None
+    component_platform: RuntimeProcessorType = None
     _file_types: List[str] = None
 
-    @property
-    def component_platform(self) -> RuntimeProcessorType:
-        return self._component_platform
+    def __new__(cls, platform_type: str):
+        parser_class_map = {subclass.component_platform.name: subclass for subclass in cls.__subclasses__()}
+        parser_subclass = parser_class_map[platform_type]
+        return super(ComponentParser, parser_subclass).__new__(parser_subclass)
 
```

Comment:
This probably warrants a comment

Reply:
Although this may be the "python" way (I really don't know), I would recommend a class factory pattern where you have a `get_instance()` `@classmethod`.  The content of this method would be similar, although I think the creation of the `parser_class_map` should occur at the class level (and be treated as a class variable).  That way it's only initialized once.

The code in `_get_component_parser_class`  would then be `ComponentParser.get_instance(platform)` - which is more clear that its acting as a class factory rather than just constructing a base `ComponentParser` instance.  (And you probably don't need `_get_component_parser_class`, since `get_instance` could perform the error handling.  Hmm, I think it might be better to raise an error rather than return `None` since the caller would always need to check for `None` and this seems fairly critical - but I'm also a bit ignorant to the bigger picture right now.)

Reply:
Updated in the latest set of commits!